.clang-tidy

```diff
@@ -1,19 +1,4 @@
-Checks:          'abseil-*,
-                  bugprone-*,
-                  clang-analyzer-*,
```

Comment:
Are we regressing anything by eliminating these?

Reply:
Those are checked but not enforced before but I don't think anyone is looking at it. So just check what we are enforcing. Then all errors are written to the fix yaml file so we can check in bash easily.

Reply:
Once we get this fixed is it possible to do a complete master run and fix all the regressions? I'm sure there are a lot. Thanks for doing this. Very excited to get this fixed!